src/agents/coordinator_agent.py

```diff
@@ -80,21 +83,39 @@ class CheckParkingOffers(CyclicBehaviour):
         """Behaviour for checking parking offers"""
 
         def _check_if_request_is_within_region(self, request_x, request_y):
-            return self.agent._x_min <= request_x < self.agent._x_max and self.agent._y_min <= request_y < self.agent._y_max
+            return (
+                self.agent._x_min <= request_x < self.agent._x_max
+                and self.agent._y_min <= request_y < self.agent._y_max
+            )
 
         async def run(self):
-            msg = await self.receive(timeout=MESSAGE_TIMEOUT)
+            # msg = await self.receive(timeout=MESSAGE_TIMEOUT)
+            msg = Message(
+                to="regionalcoordinator1@{DEFAULT_HOST}",
+                body=json.dumps(CheckOffers(0, 1, 0, 1).dict()),
+                thread="user1@{DEFAULT_HOST}",
+                metadata={"performative": "query-ref", "action": "check-parking"},
+            )
             if msg:
                 check_offers_message = CheckOffers(**json.loads(msg.body))
-                if self._check_if_request_is_within_region(check_offers_message.x, check_offers_message.y):
-                    self.agent._per_user_data[check_offers_message.sender] = {
-                        "parkings": [], "destination": (check_offers_message.x, check_offers_message.y)}
-                    check_parking = str(CheckParking(check_offers_message.time_start,
-                                        check_offers_message.time_stop).dict())
+                if self._check_if_request_is_within_region(int(check_offers_message.x), int(check_offers_message.y)):
+                    # commented to simplify tests
+                    # self.agent._per_user_data[check_offers_message.sender] = {
+                    #     "parkings": [],
+                    #     "destination": (check_offers_message.x, check_offers_message.y),
+                    # }
+                    check_parking = str(
+                        CheckParking(check_offers_message.time_start, check_offers_message.time_stop).dict()
+                    )
                     for jid in self.agent._parking_agents_jids:
+                        print(f"Sending to jid: {jid}")
                         # user_jid@host in message.thread, need in response from parking agent
-                        to_send = Message(to=jid, body=check_parking, thread=check_offers_message.sender, metadata={
-                                          "performative": "query-ref", "action": "check-parking"})
+                        to_send = Message(
+                            to=jid,
+                            body=check_parking,
+                            thread=check_offers_message.sender,
```

Comment:
Right now this `thread` value does not work. Setting it to `f"regionalcoordinator1@{DEFAULT_HOST}"` results in the following logs output:
```
2024-12-02 22:19:11 Sending to jid: parking100@server_hello
2024-12-02 22:19:11 No behaviour matched for message: <message to="parking100@server_hello" from="regional_coordinator1@server_hello" thread="regionalcoordinator1@server_hello" metadata={'performative': 'query-ref', 'action': 'check-parking'}>
2024-12-02 22:19:11 {'time_start': '0', 'time_stop': '1'}
2024-12-02 22:19:11 </message>
```

Reply:
in thread ParkingAgent should place user_jid, because regionalCoordinator has to know where to send offers